.github/workflows/auto-update.js

```diff
@@ -108,13 +108,26 @@ module.exports = async ({github, context, core, require}) => {
     url = ${next_url};
     sha256 = "${next_sha256}";`);
 
-    writeFileSync(source_path, next_source);
-
     const [_full_match, _channel_variant, _old_date, old_git_sha] =
         old_source.match(source_regex);
     const url =
         `https://github.com/NixOS/nixpkgs/compare/${old_git_sha}...${next_sha}`;
 
+    // Only write the file if the commit hash has changed
+    // Otherwise just cancel the workflow. We don't need to do anything
+    if (next_sha.startsWith(old_git_sha)) {
```

Comment:
`next_sha` will always be a full SHA. Sometimes I update sources manually and add a short SHA. this detects these cases.